Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <SdfAssetBuilder/SdfAssetBuilder.h>
+
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/IO/IOUtils.h>
+#include <AzCore/Serialization/Json/JsonUtils.h>
+#include <AzFramework/StringFunc/StringFunc.h>
+#include <AzToolsFramework/Entity/EntityUtilityComponent.h>
+#include <AzToolsFramework/Prefab/PrefabLoaderScriptingBus.h>
+#include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
+#include <AzToolsFramework/Prefab/PrefabSystemScriptingBus.h>
+#include <AzToolsFramework/Prefab/Procedural/ProceduralPrefabAsset.h>
+
+#include <AssetBuilderSDK/AssetBuilderSDK.h>
+#include <AssetBuilderSDK/SerializationDependencies.h>
+
+namespace ROS2
+{
+        namespace
+        {
+            [[maybe_unused]] constexpr const char* SdfAssetBuilderName = "SdfAssetBuilder";
+            constexpr const char* SdfAssetBuilderJobKey = "Sdf Asset Builder";
+            constexpr AZStd::array<const char*,4> SdfAssetSourceExtensions =
+            {
+                "*.sdf",
+                "*.urdf",
+                "*.world",
+                "*.xacro"
+            };
+        }
+
+    void SdfAssetBuilder::RegisterBuilder()
+    {
+        AssetBuilderSDK::AssetBuilderDesc sdfAssetBuilderDescriptor;
+
+        sdfAssetBuilderDescriptor.m_name = SdfAssetBuilderJobKey;
+        sdfAssetBuilderDescriptor.m_version = 1; // bump this to rebuild all sdf files
+        sdfAssetBuilderDescriptor.m_busId = azrtti_typeid<SdfAssetBuilder>();
+
+        for(const auto& extension : SdfAssetSourceExtensions)
+        {
+            sdfAssetBuilderDescriptor.m_patterns.push_back(AssetBuilderSDK::AssetBuilderPattern(extension, AssetBuilderSDK::AssetBuilderPattern::PatternType::Wildcard));
+        }
+
+        sdfAssetBuilderDescriptor.m_createJobFunction = AZStd::bind(&SdfAssetBuilder::CreateJobs, this,
+            AZStd::placeholders::_1, AZStd::placeholders::_2);
+        sdfAssetBuilderDescriptor.m_processJobFunction = AZStd::bind(&SdfAssetBuilder::ProcessJob, this,
+            AZStd::placeholders::_1, AZStd::placeholders::_2);
```

Comment:
Lambdas should be preferred over AZStd::bind, as it is quicker to compile and call and safer to use as it doens't allow for ignoring placeholder parameters: https://clang.llvm.org/extra/clang-tidy/checks/modernize/avoid-bind.html

Reply:
Fixed!